src/game_objects/grid.cpp

```diff
@@ -115,3 +107,52 @@ void Grid::move_selection(Direction d) {
         break;
     }
 }
+
+void Grid::load_map(const char *map_loc) {
+    // Loads map from file
+    std::ifstream fin;
+    fin.open(map_loc);
+
+    // File does not exist, exit
+    if (!fin.good())
+        return;
+
+    std::string str_buf;
+    int int_buf;
+
+    // Parse File
+    while (!fin.eof()) {
+        fin >> str_buf;
+        if (str_buf == "height") {
+            fin >> int_buf;
+            height = int_buf;
+        }
+        else if (str_buf == "width") {
+            fin >> int_buf;
+            width = int_buf;
+        }
+        else if (str_buf == "tag") {
+            // To implement later, if we want some automated way
+            // in the file to check what number corresponds to what type of tile
+        }
+        else if (str_buf == "data") {
+            for (int r = 0; r < height; r++) {
+                std::vector<Tile *> new_row;
+
+                for (int c = 0; c < width; c++) {
+                    fin >> int_buf;
+                    if (int_buf == 3) {
+                        FloorTile *new_tile = new FloorTile(c, r);
```

Comment:
Maybe we can pull this out to a `Tile *make_tile(int tile_id, int c, int r)` so we can do `new_row.push_back(make_tile(int_buf, c, r))`?